taskfile/task.go

```diff
@@ -115,12 +145,66 @@ func (t *Task) UnmarshalYAML(node *yaml.Node) error {
 		t.Prefix = task.Prefix
 		t.IgnoreError = task.IgnoreError
 		t.Run = task.Run
+		on, err := processPlatforms(task.On)
+		if err != nil {
+			return fmt.Errorf("yaml: line %d: %v", node.Line, err)
+		}
+		t.On = on
 		return nil
 	}
 
 	return fmt.Errorf("yaml: line %d: cannot unmarshal %s into task", node.Line, node.ShortTag())
 }
 
+// supportedOSes is a list of supported OSes
+var supportedOSes = map[string]struct{}{
+	"windows": {},
+	"darwin":  {},
+	"linux":   {},
+	"freebsd": {},
+}
+
+// supportedArchs is a list of supported architectures
+var supportedArchs = map[string]struct{}{
+	"amd64": {},
+	"arm64": {},
+	"386":   {},
+}
+
+// processPlatforms iterates the given platform strings,
+// validates the values and returns a slice of platform data.
+func processPlatforms(on []string) ([]*Platform, error) {
+
+	var platforms []*Platform
+
+	for _, onValue := range on {
+		splitValues := strings.Split(onValue, "/")
+		if len(splitValues) > 2 {
+			return nil, fmt.Errorf("invalid OS/Arch provided: %s", onValue)
```

Comment:
The errors from this method get bubbled up and formatted as you proposed in the calling method. The reason for doing this is because one caller has the node information and one doesn't so this way would keep the errors consistent. I'm happy to change it though.